pkg/knowledgebase/properties/bool_property.go

```diff
@@ -84,6 +84,9 @@ func (b *BoolProperty) Type() string {
 }
 
 func (b *BoolProperty) Validate(resource *construct.Resource, value any, ctx knowledgebase.DynamicContext) error {
+	if b.DeployTime && value == nil {
+		return nil
+	}
```

Comment:
same for other properties, probably want to check if the property is required + resource is imported and then ensure its not nil